### PR TITLE
Adding dateutil module with some useful datetime-related methods:

### PR DESCRIPTION
**Pull request by jkochaniak**
parse_iso8601_utc_datetime
datetime_to_epoch_seconds

### CODE REVIEW

**Comment by jkochaniak** [comment]
With these changes, I'm getting the following error running `tox` with the `py35` env:
```
_______________________________________________________________________________ ERROR collecting tests/test_spreadsheet.py _______________________________________________________________________________
tests/test_spreadsheet.py:7: in <module>
    from upside_core.spreadsheet import Spreadsheet
upside_core/spreadsheet.py:6: in <module>
    import gdata.gauth
.tox/py35/lib/python3.5/site-packages/gdata/__init__.py:27: in <module>
    import atom
.tox/py35/lib/python3.5/site-packages/atom/__init__.py:133: in <module>
    CreateClassFromXMLString)
.tox/py35/lib/python3.5/site-packages/atom/__init__.py:97: in mark_deprecated
    optional_warn_function.func_name = f.func_name
E   AttributeError: 'function' object has no attribute 'func_name'
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 1 errors during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
======================================================================================== 1 error in 1.16 seconds =========================================================================================
ERROR: InvocationError: '/Users/joanna/opt/upside/upside-core/.tox/py35/bin/py.test tests/ --cov upside_core --cov-report term-missing --cov-report xml'
________________________________________________________________________________________________ summary _________________________________________________________________________________________________
  py27: commands succeeded
ERROR:   py35: commands failed
```

@rickmcphee any ideas?

File: upside_core/dateutil.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import pytz
+import re
```
Comment (bmcniel) [inline upside_core/dateutil.py]: I use python-datetuil for this everywhere else.

**Comment by bmcniel** [comment]
Overall this looks good except I use python-dateutil for the timezone bits. It might even have utility functions that cover all this work. 

https://pypi.python.org/pypi/python-dateutil

**Comment by jkochaniak** [comment]
I found `python-dateutil` today after already writing this, but I figured since there are places already using `pytz` this would still be useful. But I'll switch those to use `dateutil` instead. Do you know of a function in `dateutil` to get epoch seconds? I didn't see one when I looked through the docs.

**Comment by bmcniel** [comment]
time.time() is epoch seconds

https://docs.python.org/2/library/time.html#time.time

File: tests/test_spreadsheet.py [hunk 2]
```diff
@@ -1,10 +1,10 @@
 try:
     # Python 2
     from StringIO import StringIO
-    from upside_core.spreadsheet import Spreadsheet
 except ImportError:
     # Python 3
     from io import StringIO
+from upside_core.spreadsheet import Spreadsheet
```
Comment (rickmcphee) [inline tests/test_spreadsheet.py]: The gdata package is not compatible with python3.  It's time to use the new sheets apis that @bmcniel has been using.  In the meantime, we can add a directive to not run the test_spreadsheet test in python3.  Will look into that fix.

**Comment by jkochaniak** [comment]
I figured out how to reorganize my gordon project (app-metrics-lambda) to allow me to share code between lambdas, so for now I'm just adding these util methods there.

**Pull request closed without merging**